src/index.ts

```diff
@@ -190,8 +217,11 @@ export const runDev = (
         cfg.ignore.some(isRegExpMatch(required))
 
       if (!isIgnored && (cfg.deps === -1 || getLevel(required) <= cfg.deps)) {
+        requiredFiles[required] = true
         log.debug(required, 'added to watcher')
-        watcher.add(required)
+        // Individual file watchers have issues in Docker containers.
+        // When not using polling, watch parent directories instead.
+        watcher.add(opts.poll ? required : path.dirname(required))
```

Comment:
This is the key change -- we want to watch the directory that contains the file.